crypto/accumulator/merkletree/readers.go

```diff
@@ -0,0 +1,72 @@
+// from https://gitlab.com/NebulousLabs/merkletree
```

Comment:
you need to include the licence header from NebulousLabs here. If you can use their repo as a go mod dependency that would be cleaner (ie if code is not modified)
`
// Original Copyright 2015 Nebulous
// MIT licence header
// Copyright 2020 ConsenSys AG
// Apache licence

Reply:
I needed to modify their code: originally the leaf sum and node sum in the Merkle tree were done differently for domain separation purpose. The thing is the way it is done (according to  the specs) makes it snark unfriendly, it adds a lot of constraints. So after discussion with the applied research team we decided to find an easier way to differentiate the leaf node from the node sum (currently it's the same, but a future solution would be to have 2 mimc with different seeds for the leaf and the node).